python-packages/order_utils/stubs/web3/utils/datatypes.pyi

```diff
@@ -1,3 +1,16 @@
+from typing import Any
+
+
+class ContractFunction:
+    @staticmethod
+    def call() -> Any: ...
+    ...
+
+
 class Contract:
-    def call(self): ...
+    class functions:
+        def __getattr__(self, function_name) -> ContractFunction: ...
+        @staticmethod
+        def isValidSignature(data: str, signer: str, signature: str) -> ContractFunction: ...
```

Comment:
Does the type checking fail if you remove this? It doesn't seem right to have a contract-specific method listed in this contract-agnostic API.

Reply:
i think it failed earlier. ill try again